### PR TITLE
Ignore warning caused by fastrlock

### LINKED ISSUE #2476: `import cupy` causes `ImportWarning`
Reported [here](https://github.com/chainer/chainer/pull/8131#issuecomment-532521654).

(by running with `python -W error`)
> ```
> >>> import cupy
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "******/python3.7/site-packages/cupy/__init__.py", line 12, in <module>
>     from cupy import core  # NOQA
>   File "******/python3.7/site-packages/cupy/core/__init__.py", line 1, in <module>
>     from cupy.core import core  # NOQA
>   File "cupy/core/core.pyx", line 1, in init cupy.core.core
>   File "******/python3.7/site-packages/cupy/cuda/__init__.py", line 4, in <module>
>     from cupy.cuda import compiler  # NOQA
>   File "******/python3.7/site-packages/cupy/cuda/compiler.py", line 12, in <module>
>     from cupy.cuda import function
>   File "cupy/cuda/memory.pxd", line 11, in init cupy.cuda.function
>   File "cupy/cuda/memory.pyx", line 1, in init cupy.cuda.memory
>   File "fastrlock/_lock.pxi", line 4, in init fastrlock.rlock
> ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
> ```

It is actually `fastrlock` that is causing `ImportWarning`:
```
$ python -W error -c 'import fastrlock.rlock'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "fastrlock/_lock.pxi", line 4, in init fastrlock.rlock
ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
```

### PR DESCRIPTION
**Pull request by emcastillo**
Closes #2476
```
$ python -W error
>>> import cupy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ecastill/em-cupy/cupy/__init__.py", line 12, in <module>
    from cupy import core  # NOQA
  File "/home/ecastill/em-cupy/cupy/core/__init__.py", line 1, in <module>
    from cupy.core import core  # NOQA
  File "cupy/core/core.pyx", line 1, in init cupy.core.core
    # distutils: language = c++
  File "/home/ecastill/em-cupy/cupy/cuda/__init__.py", line 4, in <module>
    from cupy.cuda import compiler  # NOQA
  File "/home/ecastill/em-cupy/cupy/cuda/compiler.py", line 12, in <module>
    from cupy.cuda import function
  File "cupy/cuda/memory.pxd", line 12, in init cupy.cuda.function

  File "cupy/cuda/memory.pyx", line 1, in init cupy.cuda.memory
    # distutils: language = c++
  File "fastrlock/_lock.pxi", line 4, in init fastrlock.rlock
ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
```

The problem is on the code generated by Cython when compiling memory.pyx, its introduced by distutils when it add the stubs to call fastrlock
The PR silences it right before importing the main offending module.

### CODE REVIEW

File: cupy/__init__.py [hunk 1]
```diff
@@ -9,7 +9,10 @@
 
 
 try:
-    from cupy import core  # NOQA
+    with warnings.catch_warnings():
+        distutils_warning = 'can\'t resolve package from __spec__'
+        warnings.filterwarnings("ignore", message=distutils_warning)
```
Comment (niboshi) [inline cupy/__init__.py]: Use single quotes for strings.

Comment (niboshi) [inline cupy/__init__.py]: Why not just write the message in-line?

Comment (niboshi) [inline cupy/__init__.py]: Specify `category` argument.

**Comment by emcastillo** [comment]
PTAL

**Comment by niboshi** [comment]
Did you forget to push the change?

**Comment by emcastillo** [comment]
the git pull was stuck lol

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**